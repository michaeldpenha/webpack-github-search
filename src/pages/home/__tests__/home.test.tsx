import React from 'react';
import { GetSearchDocument, SearchType } from 'types/schema';
import { act } from 'react-dom/test-utils';
import IssueSearchResponse from '../../../../__mocks__/issue-search.json';
import routeData from 'react-router';
import renderPageEl from 'src/testUtils/tests-page-render';
import Home from '..';

const mockLocation = {
  pathname: '/',
  hash: '',
  search: '',
  state: '',
};

const mocks = [
  {
    request: {
      query: GetSearchDocument,
      variables: {
        first: 15,
        type: 'ISSUE',
        query: ' type:issue language:all state:open',
      },
    },
    result: IssueSearchResponse,
  },
];

beforeEach(() => {
  jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
});

const setup = () => renderPageEl(<Home />, '/', '/', mocks);

async function wait(ms = 10) {
  await act(() => {
    return new Promise((resolve) => {
      setTimeout(resolve, ms);
    });
  });
}

describe('testing Home', () => {
  it('renders home page', async () => {
    const { container } = setup();

    await wait();

    expect(container.textContent).toContain('Github');
    expect(container).toMatchSnapshot();
  });
});
