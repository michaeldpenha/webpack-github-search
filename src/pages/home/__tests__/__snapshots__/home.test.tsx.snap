// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home renders home page 1`] = `
<div>
  <div
    class="styles__Row-kmokpt-0 gljyHy"
  >
    <div
      class="styles__Column-kmokpt-1 styles__SearchContainer-kmokpt-3 haTjGN gKVgvj"
    >
      <label
        class="styles__Label-kmokpt-5 iQCvuR"
      >
        Search
      </label>
      <input
        autocomplete="on"
        class="styles__Input-o83qfn-0 jZCHCW"
        data-testid="search"
        name="search"
        placeholder="Search issue by name or body"
        type="search"
        value=""
      />
    </div>
    <div
      class="styles__Column-kmokpt-1 styles__LookUpContainer-kmokpt-4 haTjGN"
    >
      <label
        class="styles__Label-kmokpt-5 iQCvuR"
      >
        Repository
      </label>
      <div
        class="styles__LookUpContainer-sc-1syvhlz-0 pySkO"
      >
        <input
          autocomplete="off"
          class="styles__Input-o83qfn-0 jZCHCW"
          data-testid="search"
          name="search"
          placeholder="Search Repo by name"
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="styles__Column-kmokpt-1 styles__DropDownContainer-kmokpt-2 haTjGN cvRwes"
    >
      <label
        class="styles__Label-kmokpt-5 iQCvuR"
      >
        State
      </label>
      <div
        class="styles__DropDownContainer-sc-1c97jj7-0 kHNWhL"
        data-testid="drop-down-container"
      >
        <div
          class="styles__DropDownHeader-sc-1c97jj7-1 dPLnHw"
          data-testid="drop-down-selected-value"
        >
          Open
        </div>
      </div>
    </div>
  </div>
  <main
    class="styles__ListContainer-kmokpt-6 gEHxsV"
  >
    <strong>
      0 Issues found for All Github repos in open state
    </strong>
    <div
      class="styles__LoaderBlock-cptm6y-0 fWwgZh"
    >
      <svg
        aria-labelledby="3dc7ze-aria"
        role="img"
        width="90vw"
      >
        <title
          id="3dc7ze-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#3dc7ze-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#3dc7ze-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="3dc7ze-diff"
          >
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="20"
            />
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="40"
            />
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="60"
            />
          </clippath>
          <lineargradient
            id="3dc7ze-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#f3f3f3"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#ecebeb"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#f3f3f3"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
    <div
      class="styles__LoaderBlock-cptm6y-0 fWwgZh"
    >
      <svg
        aria-labelledby="8cz3l7-aria"
        role="img"
        width="90vw"
      >
        <title
          id="8cz3l7-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#8cz3l7-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#8cz3l7-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="8cz3l7-diff"
          >
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="20"
            />
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="40"
            />
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="60"
            />
          </clippath>
          <lineargradient
            id="8cz3l7-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#f3f3f3"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#ecebeb"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#f3f3f3"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
    <div
      class="styles__LoaderBlock-cptm6y-0 fWwgZh"
    >
      <svg
        aria-labelledby="hf4vi4a-aria"
        role="img"
        width="90vw"
      >
        <title
          id="hf4vi4a-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#hf4vi4a-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#hf4vi4a-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="hf4vi4a-diff"
          >
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="20"
            />
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="40"
            />
            <rect
              height="12"
              rx="5"
              ry="5"
              width="100%"
              x="10"
              y="60"
            />
          </clippath>
          <lineargradient
            id="hf4vi4a-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#f3f3f3"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#ecebeb"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#f3f3f3"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </main>
</div>
`;
